llvm/lib/ExecutionEngine/Orc/TargetProcess/JITLoaderVTune.cpp

```diff
@@ -0,0 +1,218 @@
+//===------- JITLoaderVTune.cpp - Register profiler objects -----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Register objects for access by profilers via the VTune JIT interface.
+//===----------------------------------------------------------------------===//
+
+#include "llvm/ExecutionEngine/Orc/TargetProcess/JITLoaderVTune.h"
+#include "llvm/ExecutionEngine/Orc/Shared/VTuneSharedStructs.h"
+#include <map>
+
+#if LLVM_USE_INTEL_JITEVENTS
+#include "IntelJITEventsWrapper.h"
+#include "ittnotify.h"
+
+using namespace llvm;
+using namespace llvm::orc;
+
+static std::unique_ptr<IntelJITEventsWrapper> Wrapper;
```

Comment:
Can this global variable be placed into a context class?

Reply:
Hi Fangrui, I fixed it into a context class, but I do not see the reason why we need a context class here because `Wrapper` is only used in `JITLoaderVTune.cpp`. Could you please elaborate more on it? Thank you!

Reply:
The idea is to avoid a global state, more preferrable if the state can be shared between multiple threads.

Reply:
Thank you! Fixed! Please let me know if there is anything else I need to change.